proposals/features-composite-dependencies.md

```diff
@@ -0,0 +1,117 @@
+| `version` | `string` | The version of the Feature.  This follows the same semantics of the `version` property in the `devcontainer-feature.json` file. |
+| `features` | `array` | An array of objects (in installation order) that define the Feature(s) that compose this Feature. |
+| `features.id` | `string` | The ID of the Feature that this Feature depends on. |
+| `features.version` | `string` | The version of the Feature that this Feature depends on. |
+| `features.detect` | `string` | A command that will be executed in a shell to determine if the Feature should be installed.  If the command returns a non-zero exit code, the Feature will be installed.  If the command returns a zero exit code, the remaining install steps will be skipped. |
+| `features.options` | `object` | An object of key/value pairs that will be passed to the Feature's `install.sh` script. |
+
+#### Example `devcontainer-feature.composite.json`
+
+```jsonc
+{
+    "id": "ghcr.io/devcontainers/features/composite",
+    "version": "1.0.0",
+    "features": [
+        {
+            "id": "ghcr.io/devcontainers/features/a", // Must be a standalone Feature. (A composite Feature cannot depend on a composite Feature).
+            "version": "1.2.3",  // An exact version is required.  We do not permit pinning to a major or minor version.
+            "detect": "a --version && cat /etc/a/.markerfile", // Only continue installation of this Feature if detect returns non-zero
+            "options": {
+                "bar": true
+            }
+        },
+        {
+            "id": "ghcr.io/microsoft/features/b",
+            "version": "sha256:45b23dee08af5e43a7fea6c4cf9c25ccf269ee113168c19722f87876677c5cb2", // SHA of the published artifact is OK
+            "detect": undefined, // Omit or set as 'undefined' to always install this Feature.
+            "options": {
+                "zip": "zap"
+            }
+        }
+    ]
+}
+```
+
+An optional `finalize.sh` script can be included, and will be fired after all Features have been installed.
+
+A composite Feature will be published following the same process as an [standalone dev container Feature](https://containers.dev/implementors/features) into the same namespace - following the pattern outlined in [the Features distribution spec](https://containers.dev/implementors/features-distribution/). Dependencies of a composite Feature can be published to the same or different namespaces.
+
+An example repository structure for a repo with one composite Feature and an standalone Feature can be found below:
+
+```
+$ tree 
+
+├── src
+│   ├── composite
+│   │   ├── README.md
+│   │   ├── devcontainer-feature.composite.json
+│   │   ├── finalize.sh
+│   ├── a
+│   │   ├── README.md
+│   │   ├── devcontainer-feature.json
+│   │   └── install.sh
+...
+```
+
+### Notes:
+
+- Composite Features cannot depend on other composite Features. This is to prevent a circular dependencies and deep dependency chains.
```

Comment:
```suggestion
- Composite Features cannot depend on other composite Features. This is to prevent circular dependencies and deep dependency chains.
```